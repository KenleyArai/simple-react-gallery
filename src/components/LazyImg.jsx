import React from "react";
import Lazyload from "react-lazyload";
import PropTypes from "prop-types";

import Img from "containers/Img";

const LazyImg = ({ src, srcSet, theme }) => {
  if (src) {
    return (
      <Lazyload throttle={500} height={300}>
        <Img src={src} />
      </Lazyload>
    );
  }
  return (
    <Lazyload throttle={500} height={300}>
      <Img srcSet={src} />
    </Lazyload>
  );
};

LazyImg.propTypes = {
  src: PropTypes.string,
  srcSet: PropTypes.string
};

LazyImg.defaultProps = {
  src: "",
  srcSet: ""
};

export default LazyImg;
