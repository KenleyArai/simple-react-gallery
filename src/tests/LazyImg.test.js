import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import "jest-styled-components";
configure({ adapter: new Adapter() });

import LazyImg from "components/LazyImg";
import testSrc from "../logo-small.svg";

describe("LazyImg", () => {
  it("should render without src and srcSet", () => {
    const wrapper = shallow(<LazyImg />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with src", () => {
    const wrapper = shallow(<LazyImg src={testSrc} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render with srcSet", () => {
    const wrapper = shallow(<LazyImg srcSet={testSrc} />);
    expect(wrapper).toMatchSnapshot();
  });
});
