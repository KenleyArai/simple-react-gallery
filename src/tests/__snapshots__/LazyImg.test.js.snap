// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImg should render with src 1`] = `
<LazyLoad
  height={300}
  offset={0}
  once={false}
  overflow={false}
  resize={false}
  scroll={true}
  throttle={500}
  unmountIfInvisible={false}
>
  <Img
    src="logo-small.svg"
    srcSet=""
  />
</LazyLoad>
`;

exports[`LazyImg should render with srcSet 1`] = `
<LazyLoad
  height={300}
  offset={0}
  once={false}
  overflow={false}
  resize={false}
  scroll={true}
  throttle={500}
  unmountIfInvisible={false}
>
  <Img
    src=""
    srcSet=""
  />
</LazyLoad>
`;

exports[`LazyImg should render without src 1`] = `
<LazyLoad
  height={300}
  offset={0}
  once={false}
  overflow={false}
  resize={false}
  scroll={true}
  throttle={500}
  unmountIfInvisible={false}
>
  <Img
    src=""
    srcSet=""
  />
</LazyLoad>
`;
